cmd/parallel-manager.go

```diff
@@ -182,12 +221,20 @@ func (p *ParallelManager) stopAndWait() {
 
 // newParallelManager starts new workers waiting for executing tasks
 func newParallelManager(resultCh chan URLs) *ParallelManager {
+
+	var maxMem float64
+	memStats, err := mem.VirtualMemory()
+	if err == nil {
+		maxMem = float64(memStats.Available) * 0.8 // use upto 80% of available memory.
```

Comment:
This will account for the default `GOGC:100`.
```suggestion
		maxMem = float64(memStats.Available) / 2 // use upto 50% of available memory.
```